pkg/kv/kvserver/logstore/logstore.go

```diff
@@ -517,9 +517,8 @@ func Compact(
 	}
 	value.InitChecksum(key)
 
-	if _, err := storage.MVCCPut(
-		ctx, readWriter, key, hlc.Timestamp{}, value,
-		storage.MVCCWriteOptions{Category: fs.ReplicationReadCategory},
+	if _, err := storage.MVCCBlindPut(
```

Comment:
The test #144257 ended up helpful. I first tried `PutUnversioned` here, and only after some debugging and this test failing I figured out what's actually written here (turns out it's `MVCCMetadata` wrapping `MVCCValue` wrapping `roachpb.Value`). In the end, I remembered that there is just the blind version of this put.